magento/pwa-studio
# Fix hook dependencies for app toast error handling

## Description

Back when we added toasts there was a bug where we would recursively add toasts when the error handlers in app were invoked. There was an effect that was being re-run based on the dependencies. Instead of fixing it at the time, we papered over it by removing the problematic dependency and adding a `disable-line` rule.

Funny enough, after the Talon refactor it seems like we can fix the dependency arrays without having to do much else. We can even remove the extra check within the `handleError`.

## Related Issue





Closes #1377 .

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Add an error to `miniCart.js` before the return and load the page. You should see the error toast.
2. Click the 'x' to clear the toast. It should not be replaced by a duplicate.
3. Refresh the page and this time let the error clear itself. It should disappear after some timeout and it should not be replaced by a duplicate.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
